### PR TITLE
need some more verification of operator processing of config change i…

### PR DESCRIPTION
**Pull request by gabemontero**
…n e2e's

### CODE REVIEW

**Comment by gabemontero** [comment]
setup failure

/retest

**Comment by gabemontero** [comment]
```
level=error msg="\t* aws_s3_bucket.ignition: Error creating S3 bucket: TooManyBuckets: You have attempted to create more buckets than allowed"
```

File: test/e2e/cluster_samples_operator_test.go [hunk 1]
```diff
@@ -683,6 +711,20 @@ func TestSpecManagementStateField(t *testing.T) {
 		dumpPod(t)
 		t.Fatalf("error updating Config %v", err)
 	}
+	err = wait.PollImmediate(1*time.Second, 3*time.Minute, func() (bool, error) {
+		cfg, err = crClient.Samples().Configs().Get(samplesapi.ConfigName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
```
Comment (bparees) [inline test/e2e/cluster_samples_operator_test.go]: and here

File: test/e2e/cluster_samples_operator_test.go [hunk 2]
```diff
@@ -700,6 +742,20 @@ func TestSpecManagementStateField(t *testing.T) {
 		dumpPod(t)
 		t.Fatalf("error updating Config %v and %#v", err, cfg)
 	}
+	err = wait.PollImmediate(1*time.Second, 3*time.Minute, func() (bool, error) {
+		cfg, err = crClient.Samples().Configs().Get(samplesapi.ConfigName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
```
Comment (bparees) [inline test/e2e/cluster_samples_operator_test.go]: and here

File: test/e2e/cluster_samples_operator_test.go [hunk 3]
```diff
@@ -637,7 +651,7 @@ func TestSpecManagementStateField(t *testing.T) {
 	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
 		cfg, err = crClient.Samples().Configs().Get(samplesapi.ConfigName, metav1.GetOptions{})
 		if err != nil {
-			return false, nil
+			return false, err
```
[the remark this answers is not in the document]
Comment (bparees) [reply test/e2e/cluster_samples_operator_test.go]: the more typical pattern we use is to store the err and continue retrying and if we timeout, then we print the last err we got.

[the remark this answers is not in the document]
Comment (gabemontero) [reply test/e2e/cluster_samples_operator_test.go]: ah good point ... I'll rework all of these

**Comment by gabemontero** [comment]
updates pushed @bparees 

thanks / ptal

**Comment by gabemontero** [comment]
ah we have at least a passing operator e2e with no install flake potholes

**Comment by gabemontero** [comment]
we have a lot of SCL not printing the usage message errors on the image eco @bparees  .... I forget, do we bug SCL about those ?

**Comment by gabemontero** [comment]
bunch of auth/user errors on e2e-aws, along with api server / network interaction type errors

**Comment by bparees** [comment]
> we have a lot of SCL not printing the usage message errors on the image eco @bparees .... I forget, do we bug SCL about those ?

@gabemontero The failures should be investigated, but they don't look like anything specific to the images themselves, it looks like the test failed to setup the openshift client successfully.

**Comment by gabemontero** [comment]
On Wed, Feb 13, 2019 at 10:32 PM Ben Parees <notifications@github.com>
wrote:

> we have a lot of SCL not printing the usage message errors on the image
> eco @bparees <https://github.com/bparees> .... I forget, do we bug SCL
> about those ?
>
> @gabemontero <https://github.com/gabemontero> The failures should be
> investigated, but they don't look like anything specific to the images
> themselves, it looks like the test failed to setup the openshift client
> successfully.
>
yeah the latest one were I that ilk ... user not set up, etc. ... noticed
that after I posted

**Pull request merged**